server.js

```diff
@@ -4040,6 +4040,81 @@ cache(function(data, match, sendBadge, request) {
       badgeData.text[1] = 'invalid';
       sendBadge(format, badgeData);
     }
+   });
+}));
+
+function nexusRequest(data, match, sendBadge, request, urlCallback, dataCallback) {
+  var scheme = match[1]; // http(s)
+  var host = match[3]; // nexus.example.com
+  var groupId = match[4]; // eg, `com.google.inject`
+  var artifactId = match[5]; // eg, `guice`
+  var format = match[6] || "gif";
+  var apiUrl = scheme + '://' + host + urlCallback(encodeURIComponent(groupId), encodeURIComponent(artifactId));
+  var badgeData = getBadgeData('nexus', data);
+  request(apiUrl, { headers: { 'Accept': 'application/json' } }, function(err, res, buffer) {
+	 if (err != null) {
+	   badgeData.text[1] = 'inaccessible';
+	   sendBadge(format, badgeData);
+	   return;
+	 }
+	 try {
+	   var parsed = JSON.parse(buffer);
+	   var version = dataCallback(parsed);
+
+	   if (version === '0' || /SNAPSHOT/.test(version)) {
+		 badgeData.colorscheme = 'orange';
+		 if (version !== '0') {
+		    badgeData.text[1] = version;
+		 } else {
+	        badgeData.text[1] = 'undefined';
+		 }
+	   } else {
+		 badgeData.colorscheme = 'blue';
+		 badgeData.text[1] = version;
+	   }
+	   sendBadge(format, badgeData);
+	 } catch(e) {
+	   badgeData.text[1] = 'invalid';
+	   sendBadge(format, badgeData);
+	 }
+  });
+}
+
+function isSnapshotVersion(version) {
+  var pattern = /(\d+\.)*\d\-SNAPSHOT/;
+  return version.match(pattern);
+}
+
+// standalone sonatype nexus installation
+camp.route(/^\/nexus\/r\/(http(s)?)\/((?:[^\/]+)(?:\/.+?)?)\/([^\/]+)\/([^\/]+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  nexusRequest(data, match, sendBadge, request, function(groupId, artifactId) {
+      return '/service/local/lucene/search?g=' + groupId + '&a=' + artifactId;
```

Comment:
i will also note that i've been running this on a private server w/o any kind of speed issues since this request has been open.

unfortunately it took way too long for this to be even looked at by ppl here and i just haven't had the time to circle back and make changes.